pyartifactory/objects/build.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+import logging
+from typing import Union
+
+import requests
+from requests import Response
+
+from pyartifactory.exception import ArtifactoryError, BuildNotFoundError
+from pyartifactory.models.build import (
+    BuildCreateRequest,
+    BuildDeleteRequest,
+    BuildDiffResponse,
+    BuildError,
+    BuildInfo,
+    BuildListResponse,
+    BuildPromotionRequest,
+    BuildPromotionResult,
+    BuildProperties,
+)
+from pyartifactory.objects.object import ArtifactoryObject
+
+logger = logging.getLogger("pyartifactory")
+
+
+class ArtifactoryBuild(ArtifactoryObject):
+    """Models an artifactory build."""
+
+    _uri = "build"
+
+    def get_build_info(
+        self,
+        build_name: str,
+        build_number: str,
+        properties: BuildProperties = BuildProperties(),
+    ) -> BuildInfo:
+        """
+        :param build_name: Build name to be retrieved
+        :param build_number: Build number to be retrieved
+        :param properties: Build properties model, for admitted values
+                           see https://jfrog.com/help/r/jfrog-rest-apis/build-info
+        :return: BuildInfo model object containing server response
+        """
+        build_name = build_name.strip("/")
+        build_number = build_number.strip("/")
+
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}/{build_number}{properties.to_query_string()}",
+            )
+            logger.debug("Build Info successfully retrieved")
+        except requests.exceptions.HTTPError as error:
+            self._raise_exception(error)
+
+        return BuildInfo(**response.json())
+
+    def create_build(self, create_build_request: BuildCreateRequest) -> None:
+        try:
+            self.get_build_info(create_build_request.name, create_build_request.number)
+        except BuildNotFoundError:
+            # other exception from get_build_info are forwarded to caller.
+            try:
+                # build does not exist, can be created here
+                _resp = self._put(f"api/{self._uri}", json=create_build_request.model_dump())
+                if _resp.status_code != 204:
```

Comment:
What are the possible error codes in case of error ? If they are > 400, they will be raised by `raise_for_status`. So no need of this

Reply:
Can you please answer this comment? This is the only one blocking the MR.